drink/test-resources/transfer.wat

```diff
@@ -0,0 +1,44 @@
+;; This transfers 100 balance to the zero account and copies the return code of this transfer to the output buffer.
+;; Also, emits a dummy event.
+(module
+	(import "seal0" "seal_deposit_event" (func $seal_deposit_event (param i32 i32 i32 i32)))
+	(import "seal0" "seal_transfer" (func $seal_transfer (param i32 i32 i32 i32) (result i32)))
+	(import "seal0" "seal_return" (func $seal_return (param i32 i32 i32)))
+	(import "env" "memory" (memory 1 1))
+
+	;; [0, 32) zero-adress
+	(data (i32.const 0)
+		"\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00"
+		"\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00"
+	)
+
+	;; [32, 40) 100 balance
+	(data (i32.const 32) "\64\00\00\00\00\00\00\00")
+
+	;; [40, 44) here we store the return code of the transfer
+
+	(func (export "deploy"))
+
+	(func (export "call")
+		(i32.store
+			(i32.const 40)
+			(call $seal_transfer
+				(i32.const 0) ;; ptr to destination address
```

Comment:
Using zero-address is incorrect as the private key for it is well-known ;P 

Doesn't matter in this context of course xD

Reply:
good practices etc :P

Reply:
removed transferring at all (it was silently failing due to insufficient contract's balance)

Reply:
and tests were passing? How's that possible? ie how was the event emitted if the txn failed?

Reply:
okay, taking that back - I have no idea why it was passing